frontend/src/components/Form.jsx

```diff
@@ -0,0 +1,204 @@
+import React, { useState } from "react";
+
+const _ = require('lodash');
+
+function Form(props) {
+    
+    const [formState, setFormField] = useState(() => setInitialState({}, props.fields));
+    const [imagePreview, setPreview] = useState(() => formState["image"]? formState["image"]["fileLink"] : "")
+
+    const imagestyle = {
+      height: "90px",
+      width: "90px",
+    };
+
+    function handleSubmit(e) {
+      
+      e.preventDefault();
+      props.callback(formState)
+    }
+
+    function setInitialState(newState, fields) {
+        _.forEach(fields, (field) => {
+            if (field.type === "TextString") {
+            newState[field.name] = {
+              hebrew: field.info.hebrew,
+              english: field.info.english,
+            };
+          } else {
+            newState[field.name] = field.info;
+          }
+        });
+        return newState
+    }
+
+    function createForm() {
+        
+        if (Object.keys(formState).length > 0) {
+            return _.map(props.fields, inputMaker);
+        } else {
+            return [];
+        }
+    }
+
+    function update(field) {
+      
+        if (field === "hebrew" || field === "english") {
+            return (e) => {
+              e.persist();
+                
+                setFormField(oldState => {
+                  
+                    oldState[e.target.id][field] = e.target.value;
+                    let newState = Object.assign({}, oldState)
+                    return newState;
+                })
+            }
+        } else if (field === "image") {
+          
+            return (e) => {
+              e.persist()
+              setFormField((oldState) => ({
+                ...oldState,
+                [field]: e.target.files[0],
+              }));
+              setPreview(URL.createObjectURL(e.target.files[0]));
+            };
+        } else {
+          
+            return (e) => {
+                e.persist()
+              if (e.target.type === "checkbox") {
+                
+                setFormField((oldState) => ({
+                ...oldState,
+                [field]: e.target.checked,
+                    }));
+              } else {
+                
+                setFormField((oldState) => {
+                  
+                  return {
+                    ...oldState,
+                    [field]: e.target.value
+                    }
+                  });
+                };
+            }
+        }
+    }
+
+    function inputMaker(field) {
+        switch (field.type) {
+          case "TextString":
```

Comment:
Why does the form need to know if it's string or TextString? what's the difference? for the form both should render a text input, that's it